createdisk.sh

```diff
@@ -15,36 +15,61 @@ INSTALL_DIR=${1:-crc-tmp-install-data}
 
 # If the user set OKD_VERSION in the environment, then use it to set BASE_OS
 OKD_VERSION=${OKD_VERSION:-none}
-destDirPrefix="crc"
 if [[ ${OKD_VERSION} != "none" ]]
 then
     BASE_OS=fedora-coreos
     destDirPrefix="crc_okd"
 fi
 BASE_OS=${BASE_OS:-rhcos}
 OPENSHIFT_VERSION=$(${JQ} -r .clusterInfo.openshiftVersion $INSTALL_DIR/crc-bundle-info.json)
+BUNDLE_TYPE=$(${JQ} -r .type $INSTALL_DIR/crc-bundle-info.json)
+destDirPrefix="crc"
+if [ ${BUNDLE_TYPE} == "microshift" ]; then
+    destDirPrefix="crc_${BUNDLE_TYPE}"
+    BASE_OS=rhel
+fi
 
 # SNC_PRODUCT_NAME: If user want to use other than default product name (crc)
 # VM_PREFIX: short VM name (set by SNC_PRODUCT_NAME) + random string generated by openshift-installer
 SNC_PRODUCT_NAME=${SNC_PRODUCT_NAME:-crc}
-VM_PREFIX=$(get_vm_prefix ${SNC_PRODUCT_NAME})
-VM_NAME="${VM_PREFIX}-master-0"
+if [ ${BUNDLE_TYPE} == "microshift" ]; then
+    VM_NAME=${SNC_PRODUCT_NAME}
+else
+    VM_PREFIX=$(get_vm_prefix ${SNC_PRODUCT_NAME})
+    VM_NAME="${VM_PREFIX}-master-0"
+fi
 
-# Get the IP of the VM
 VM_IP=$(sudo virsh domifaddr ${VM_NAME} | tail -2 | head -1 | awk '{print $4}' | cut -d/ -f1)
 
-# Remove unused images from container storage
-${SSH} core@${VM_IP} -- 'sudo crictl rmi --prune'
-
-# Disable kubelet service
-${SSH} core@${VM_IP} -- sudo systemctl disable kubelet
-
-# Stop the kubelet service so it will not reprovision the pods
-${SSH} core@${VM_IP} -- sudo systemctl stop kubelet
+if [ ${BUNDLE_TYPE} != "microshift" ]; then
+    # Remove unused images from container storage
+    ${SSH} core@${VM_IP} -- 'sudo crictl rmi --prune'
+    
+    # Disable kubelet service
+    ${SSH} core@${VM_IP} -- sudo systemctl disable kubelet
+    
+    # Stop the kubelet service so it will not reprovision the pods
+    ${SSH} core@${VM_IP} -- sudo systemctl stop kubelet
+fi
 
 # Enable the podman.socket service for API V2
 ${SSH} core@${VM_IP} -- sudo systemctl enable podman.socket
 
+if [ ${BUNDLE_TYPE} == "microshift" ]; then
+    # Pull openshift release images because as part of microshift bundle creation we
+    # don't run microshift service which fetch these image but instead service is run
+    # as part of crc so user have a fresh cluster instead something already provisioned
+    # but images we cache it as part of bundle.
+    ${SSH} core@${VM_IP} 'sudo bash -x -s' <<EOF
+     jq --raw-output '.images | to_entries | map(.value) | join("\n")' /usr/share/microshift/release/release-$(uname -i).json | xargs -n1 podman pull --authfile /etc/crio/openshift-pull-secret
+EOF
+    # Disable firewalld otherwise generated bundle have it running and each podman container
+    # which try to expose a port need to added to firewalld rule manually
+    # also in case of microshift the ports like 2222, 443, 80 ..etc need to be manually added
+    # and OCP/OKD/podman bundles have it disabled by default.
```

Comment:
Isn't it odd that microshift's firewalld configuration would block ports 443/80 by default? How am I supposed to reach applications running in a 'real' microshift installation?

Reply:
https://github.com/openshift/microshift/blob/release-4.12/docs/howto_firewall.md

Reply:
> MicroShift does not require a firewall for normal operation. 
> The following ports are optional and they should be considered for MicroShift if a firewall is enabled.

80 | TCP | HTTP port used to serve applications through the OpenShift router
-- | -- | --
443 | TCP | HTTPS port used to serve applications through the OpenShift router

Reply:
This is also documented in https://access.redhat.com/documentation/en-us/red_hat_build_of_microshift/4.12/html/installing/microshift-embed-in-rpm-ostree#accessing-microshift-cluster-remotely-as-admin_microshift-embed-in-rpm-ostree , quite unexpected that a microshift cluster is unable to communicate externally by default ;)

Reply:
Fwiw, I'd put these links as comment/justification and remove the manually written one.